core/trino-main/src/main/java/io/trino/server/ui/UiQueryResource.java

```diff
@@ -105,6 +120,34 @@ public Response getQueryInfo(@PathParam("queryId") QueryId queryId, @Context Htt
         return Response.status(Status.GONE).build();
     }
 
+    @ResourceSecurity(WEB_UI)
+    @GET
+    @Path("{queryId}/explain")
```

Comment:
Do we need a differentiation for distributed and plain explain in the URL?

Reply:
I'll extract this to a separate change